M2/Macaulay2/packages/PrimaryDecomposition/doc.m2

```diff
@@ -1,252 +1,546 @@
-document {
-     Key => {(associatedPrimes, Ideal),(associatedPrimes, MonomialIdeal)},
-     Headline => "find the associated primes of an ideal",
-     Usage => "associatedPrimes I\nass I",
-     Inputs => {
-	  "I" => Nothing => {ofClass Ideal, " in a (quotient of a) polynomial ring ", TT "R"}
-	  },
-     Outputs => {
-	  {"a list of the prime ideals in ", TT "R", " that are associated to ", TT "I"}
-	  },
-     TT "ass", " is an abbreviation for ", TT "associatedPrimes", ".",
-     PARA{},
-     "Computes the set of associated primes for the ideal ", TT "I", ".",
-     EXAMPLE {
-	  "R = ZZ/101[a..d];",
-	  "I = intersect(ideal(a^2,b),ideal(a,b,c^5),ideal(b^4,c^4))",
-	  "associatedPrimes I"
-	  },
-     EXAMPLE {
-	  "R = ZZ/7[x,y,z]/(x^2,x*y);",
-	  "I=ideal(0_R);",
-	  "associatedPrimes I"
-	  },
-     PARA{},
-     "In general, the associated primes are found using Ext modules: the 
-     associated primes of codimension ", TT "i", " of ", TT "I", " and ",
-     TT "Ext^i(R^1/I,R)", " are identical, as shown in 
-     Eisenbud-Huneke-Vasconcelos, Invent. Math. 110 (1992) 207-235.",
-     PARA{},
-     TO primaryDecomposition, " also computes the associated primes.  After doing 
-     a primaryDecomposition, calling ", TO "associatedPrimes", " requires no new computation,
-     and the list of associated primes is in the same order as 
-     the list of primary components returned by ", TO "primaryDecomposition", ".",
-     PARA{},
-     "If the ideal is ", ofClass MonomialIdeal, ", then a more efficient 
-     method is used.  This monomial ideal
-     code was written by Greg Smith and Serkan Hosten.  The above comments 
-     about primary decomposition hold in this case too.",
-     EXAMPLE lines ///
-     	  R = QQ[a..f];
-	  I = monomialIdeal ideal"abc,bcd,af3,a2cd,bd3d,adf,f5"
-	  ass I
-	  primaryDecomposition I
-     ///,
-     "The list of associated primes corresponds to the list of primary components of ", TT "I", ": the
-     ", TT "i", "-th associated prime is the radical of the ", TT "i", "-th primary component.",
-     PARA {
-	 BOLD "Original author: ", "C. Yackel, http://faculty.mercer.edu/yackel_ca/", ".",
-	 },
-     SeeAlso => {(primaryDecomposition,Ideal), 
-     	       "radical", "minimalPrimes", "topComponents", 
-	       "removeLowestDimension"}
-     }
+undocumented { "Increment" }
```

Comment:
This is a symbol exported by this package, so it should be documented.

Reply:
I personally think the symbol on its own is just a symbol, but when used as a strategy gets meaning and should be documented, but I'll defer to @jchen419 here.

Reply:
I think it's fine to document `Increment` - probably under the node of `primaryComponent` (which should be the only function that uses it. I should add that the EHV algorithm for modules uses a different type of increment strategy, which is not compatible with the way this option is currently set up).

Reply:
Done.